crates/ide_assists/src/handlers/extract_struct_from_enum_variant.rs

```diff
@@ -104,10 +106,18 @@ pub(crate) fn extract_struct_from_enum_variant(
                 });
             }
 
-            let def = create_struct_def(variant_name.clone(), &field_list, &enum_ast);
+            let indent = enum_ast.indent_level();
+            let def = create_struct_def(variant_name.clone(), &variant, &field_list, &enum_ast);
+            def.reindent_to(indent);
+
             let start_offset = &variant.parent_enum().syntax().clone();
-            ted::insert_raw(ted::Position::before(start_offset), def.syntax());
-            ted::insert_raw(ted::Position::before(start_offset), &make::tokens::blank_line());
+            ted::insert_all_raw(
+                ted::Position::before(start_offset),
+                vec![
+                    def.syntax().clone().into(),
+                    make::tokens::whitespace(&format!("\n\n{}", indent)).into(),
```

Comment:
Ideally, you'd be able to `insert_all` here, and have the auto-ws magic to do the indentation for you. Might be an interesting case to look at after this PR is merged.